auth_service/user_management/claims_repository/core/seed.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Functions to seed the repository with user claims.
+"""
+
+import logging
+
+from hexkit.protocols.dao import MultipleHitsFoundError, NoHitsFoundError
+
+from auth_service.config import Config
+from auth_service.deps import get_mongodb_config, get_mongodb_dao_factory
+from auth_service.user_management.claims_repository.core.claims import (
+    create_data_steward_claim,
+)
+from auth_service.user_management.claims_repository.core.utils import is_data_steward
+from auth_service.user_management.claims_repository.deps import (
+    get_claim_dao,
+    get_claim_dao_factory,
+    get_claim_dao_factory_config,
+)
+from auth_service.user_management.user_registry.deps import (
+    get_user_dao,
+    get_user_dao_factory,
+    get_user_dao_factory_config,
+)
+
+__all__ = ["seed_claims"]
+
+log = logging.getLogger(__name__)
+
+
+async def seed_claims(config: Config) -> None:
+    """Seed the claims repository with claims specified in the config."""
+    data_stewards = config.add_as_data_stewards
+    if not data_stewards:
+        return
+    user_dao = await get_user_dao(
+        dao_factory=get_user_dao_factory(
+            config=get_user_dao_factory_config(config),
+            dao_factory=get_mongodb_dao_factory(config=get_mongodb_config(config)),
+        )
+    )
+    claim_dao = await get_claim_dao(
+        dao_factory=get_claim_dao_factory(
+            config=get_claim_dao_factory_config(config),
+            dao_factory=get_mongodb_dao_factory(config=get_mongodb_config(config)),
+        )
+    )
+    for data_steward in data_stewards:
```

Comment:
Existing data steward roles need to be deleted. The config is currently the single source of truth.

Reply:
I was unsure whether we want to use the config as the sole mechanism to specify data stewards or whether we also want to allow adding data stewards on the fly to the database of the running service.

But yes, maybe better to make the config the "single source of truth" even if it means a restart will be necessary to enter new data stewards. This mechanism will be probably refined later anyway.

Reply:
Ok, done in 93e2bf0.

This whole method of seeding the database is a bit unclean because there is a race condition if multiple instances of the service are started at the same time. I think it should eventually be replaced by a job that can be run by an administrator independently of the service (e.g. using the existing POST endpoint or some other special endpoint, maybe with additional protection).

Reply:
Yes, that is true. An alternative is to not put it into the database at all but directly serve from the memory.